heudiconv/heuristics/reproin.py

```diff
@@ -780,6 +783,13 @@ def infotoids(seqinfos, outdir):
                     "Should have got a single session marker.  Got following: %s"
                     % ', '.join(map(repr, ses_markers))
                 )
+            if re.match("(X|x)*", ses_markers):
+               # replace X* or x* sessions with prior_sessions+1
+               prior_sessions = sorted(glob(os.path.join(sessions_dir, 'ses-*')))
+               if prior_sessions:
+                session = '%03d' % (len(prior_sessions) + 1)
+               else:
+                session='001'
```

Comment:
I don't think we should hardcode to `%03d` since intention on formatting might have been different etc.

- do not just take the length of `prior_sessions` but
  - extract labels (what comes after `ses-`), see what length they have.  if non-uniform (`1,...,12`) - use plain `'%d'`. If uniform - use `f'%0{len(ses_labels[0])}d`'`
  - do not base on `len(prior_sessions)` but convert all session labels into `int`. If errors - errors. Take max of those ints, so smth like `max(map(int, ses_labels))`
- make this logic into a function  `get_next_xxx_session(prior_session_dirs: list[str])`, for which also add unittests for it